Code/Engine/ECS/Systems/PlayerControllerSystem.cpp

```diff
@@ -0,0 +1,116 @@
+#include "sadpch.h"
+
+#include <SDL2/SDL.h>
+
+#include <Engine/InputManager.h>
+#include <Engine/Transform.h>
+
+#include <Engine/ECS/Registry.h>
+#include <Engine/ECS/Systems/PlayerControllerSystem.h>
+#include <Engine/ECS/Components/TransformComponent.h>
+#include <Engine/ECS/Components/PlayerControllerComponent.h>
+#include <Engine/ECS/Components/EditorControllerComponent.h>
+
+sad::PlayerControllerSystem::PlayerControllerSystem() {}
+sad::PlayerControllerSystem::~PlayerControllerSystem() {}
+
+void sad::PlayerControllerSystem::PlayerControls(InputManager& input, const ecs::TransformComponent& transformComponent, float movespeed)
+{
+	bool usingController = false; // bool that disables movement keys when joystick being used to prevent double speed
```

Comment:
Does this have to be determined by the system? Could we query this from the input manager - where we just toggle it whenever a controller/keyboard event is detected? 

For instance...
```
if (input.IsControllerActive())
{
    // Check controller inputs
}
else
{
    // Check other inputs
}
```

We already use an `m_ControllerIsActive` boolean in the InputManager which could be leveraged for this... (we could check for duplicate controllers in the `InputManager::OnControllerConnected` by checking `if (m_Controller)` instead)
```
void CatchControllerEvents(...)
{
    if (sdl detects controller event)
    {
        m_ControllerIsActive = true;
    }
}

void CatchKeyboardEvents(...)
{
    if (sdl detects keyboard event)
    {
        m_ControllerIsActive = false;
    }
}

bool IsControllerActive() { return m_ControllerIsActive; }
```